Source/WebCore/platform/graphics/coretext/FontPlatformDataCoreText.cpp

```diff
@@ -183,4 +184,16 @@ String FontPlatformData::familyName() const
     return { };
 }
 
+FontPlatformData FontPlatformData::cloneWithSize(const FontPlatformData& source, float size)
+{
+    FontPlatformData copy(source);
+    copy.m_size = size;
+    // We need to reinitialize the instance, because the difference in size
+    // necessitates a new scaled font instance.
+    ASSERT(copy.m_font.get());
+    copy.m_font = adoptCF(CTFontCreateCopyWithAttributes(copy.m_font.get(), copy.m_size, nullptr, CTFontCopyFontDescriptor(copy.m_font.get())));
```

Comment:
This leaks the result of `CTFontCopyFontDescriptor`. We have to use `adoptCF` to prevent that. But also pretty sure we don’t need to pass the font descriptor from the font itself with all the attributes the font already has. The descriptor is only needed for additional attributes that we want to be different from the ones already in the font. So we should just be passing `nullptr`.

We also have to set `copy.m_ctFont` to nullptr.

Reply:
Agreed, I think we should try to implement the `font-size-adjust` property without cloning any fonts, and without creating any additional font objects.